Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -82,39 +98,112 @@ public Registry(Uri baseUri)
+        }
+        var matchingManifest = ridDict[bestManifestRid];
+        var manifestResponse = await GetManifest(repositoryName, matchingManifest.digest);
+        return await TryReadSingleImage(repositoryName, await manifestResponse.Content.ReadFromJsonAsync<ManifestV2>());
+    }
 
-        if (manifest is null) throw new NotImplementedException("Got a manifest but it was null");
+    private async Task<HttpResponseMessage> GetManifest(string repositoryName, string reference)
+    {
+        var client = GetClient();
+        var response = await client.GetAsync(new Uri(BaseUri, $"/v2/{repositoryName}/manifests/{reference}"));
+        response.EnsureSuccessStatusCode();
+        return response;
+    }
 
-        if ((string?)manifest["mediaType"] != DockerManifestV2)
-        {
-            throw new NotImplementedException($"Do not understand the mediaType {manifest["mediaType"]}");
-        }
+    async Task<HttpResponseMessage> GetBlob(string repositoryName, string digest)
+    {
+        var client = GetClient();
+        var response = await client.GetAsync(new Uri(BaseUri, $"/v2/{repositoryName}/blobs/{digest}"));
+        response.EnsureSuccessStatusCode();
+        return response;
+    }
 
-        JsonNode? config = manifest["config"];
-        Debug.Assert(config is not null);
-        Debug.Assert(((string?)config["mediaType"]) == DockerContainerV1);
+    private string? CheckIfRidExistsInGraph(RuntimeGraph graphForManifestList, string userRid) => graphForManifestList.Runtimes.FirstOrDefault(kvp => graphForManifestList.AreCompatible(kvp.Key, userRid)).Key;
 
-        string? configSha = (string?)config["digest"];
-        Debug.Assert(configSha is not null);
+    private (IReadOnlyDictionary<string, PlatformSpecificManifest>, RuntimeGraph) ConstructRuntimeGraphForManifestList(ManifestListV2 manifestList, RuntimeGraph dotnetRuntimeGraph)
+    {
+        var ridDict = new Dictionary<string, PlatformSpecificManifest>();
+        var runtimeDescriptionSet = new HashSet<RuntimeDescription>();
+        foreach (var manifest in manifestList.manifests) {
+            if (CreateRidForPlatform(manifest.platform) is { } rid)
+            {
+                if (ridDict.TryAdd(rid, manifest)) {
+                    AddRidAndDescendantsToSet(runtimeDescriptionSet, rid, dotnetRuntimeGraph);
+                }
+            }
+        }
+        
+        var graph = new RuntimeGraph(runtimeDescriptionSet);
+        return (ridDict, graph);
+    }
 
-        response = await client.GetAsync(new Uri(BaseUri, $"/v2/{name}/blobs/{configSha}"));
+    private string? CreateRidForPlatform(PlatformInformation platform)
+    {
+        var osPart = platform.os switch
+        {
+            "linux" => "linux",
+            "windows" => "win",
+            _ => null
```

Comment:
Should this be `null` and continue, or throw? I expected throw.

Reply:
Ah, so this is "if we hit something we don't know about ignore it and hope another entry is known to us"? That seems reasonable (maybe a comment though).